pkg/source/firefly/source.go

```diff
@@ -0,0 +1,156 @@
+package firefly
+
+import (
+	"context"
+	"strings"
+
+	pbtypes "github.com/gogo/protobuf/types"
+	"github.com/spf13/pflag"
+	"go.thethings.network/lorawan-stack/v3/pkg/ttnpb"
+	"go.thethings.network/lorawan-stack/v3/pkg/types"
+
+	"go.thethings.network/lorawan-stack-migrate/pkg/source"
+	"go.thethings.network/lorawan-stack-migrate/pkg/source/firefly/api"
+	"go.thethings.network/lorawan-stack-migrate/pkg/source/firefly/devices"
+)
+
+type Source struct {
+	ctx context.Context
+
+	config *config
+}
+
+// NewSource creates a ner Firefly source
+func NewSource(ctx context.Context, flags *pflag.FlagSet) (source.Source, error) {
+	config, err := getConfig(flags)
+	if err != nil {
+		return Source{}, err
+	}
+	return Source{
+		ctx: ctx,
+
+		config: config,
+	}, nil
+}
+
+// ExportDevice implements the source.Source interface.
+func (s Source) ExportDevice(devEUI string) (*ttnpb.EndDevice, error) {
+	ffdev, err := devices.GetDevice(devEUI)
+	if err != nil {
+		return nil, err
+	}
+	v3dev := &ttnpb.EndDevice{
+		Name:        ffdev.Name,
+		Description: ffdev.Description,
+		// Formatters:      &ttnpb.MessagePayloadFormatters{},
+		FrequencyPlanId: s.config.frequencyPlanID,
+		Ids: &ttnpb.EndDeviceIdentifiers{
+			ApplicationIds: &ttnpb.ApplicationIdentifiers{ApplicationId: "firefly-" + s.config.appID},
```

Comment:
I think we shouldn't prefix this. I know that we do it in `chirpstack` but I think we should use `s.config.appID` directly.